theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -1,175 +1,163 @@
 def katello_versions = [
     'master': [
         'foreman': 'develop',
-        'ruby': '2.5',
+        'ruby': ['2.5', '2.7']
     ],
     'KATELLO-4.0': [
         'foreman': '2.4-stable',
-        'ruby': '2.5'
+        'ruby': ['2.5']
     ],
     'KATELLO-3.18': [
         'foreman': '2.3-stable',
-        'ruby': '2.5'
+        'ruby': ['2.5']
     ],
     'KATELLO-3.17': [
         'foreman': '2.2-stable',
-        'ruby': '2.5'
+        'ruby': ['2.5']
     ],
 ]
 
-def ruby = katello_versions[ghprbTargetBranch]['ruby']
+def ruby_versions = katello_versions[ghprbTargetBranch]['ruby']
 def foreman_branch = katello_versions[ghprbTargetBranch]['foreman']
 
 pipeline {
+    agent none
+
     options {
         timestamps()
         timeout(time: 3, unit: 'HOURS')
         ansiColor('xterm')
     }
 
-    agent { label 'fast' }
-
     stages {
-        stage('Setup Git Repos') {
-            steps {
-                deleteDir()
-                ghprb_git_checkout()
-
-                dir('foreman') {
-                   git url: "https://github.com/theforeman/foreman", branch: foreman_branch, poll: false
+        stage('Test') {
+            matrix {
+                agent { label 'fast' }
+                axes {
+                    axis {
+                        name 'ruby'
+                        values '2.5', '2.7'
+                    }
                 }
-            }
-        }
-        stage("Setup RVM") {
-            steps {
+                when {
+                    expression {
+                        ruby_versions.contains(env.ruby)
```

Comment:
Is this set up because you can't use variables in `values`?

Reply:
Correct.